llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -5974,6 +6023,10 @@ SDValue NVPTXTargetLowering::PerformDAGCombine(SDNode *N,
     case ISD::VSELECT:
       return PerformVSELECTCombine(N, DCI);
   }
+
+  if (N->isMachineOpcode() && N->getMachineOpcode() == NVPTX::CVT_u32_u64)
+    return PerformTruncCombine(N, DCI);
+
```

Comment:
Another special case that may not be necessary if the optimization is moved to DAGToDAG, and only deals with machine nodes.

Reply:
Resolving because the implementation has been moved to DAGCombine.